### PR TITLE
MUI Dialog

### PR DESCRIPTION
**Pull request by pushkatel**
Adds a prop `showCloseIcon` to control whether the close icon and title render on a `Dialog` window

-Default is set to true

### CODE REVIEW

**Comment by Gelio** [comment]
LGTM :smile: 

btw. this PR should be merged now that the MR in the parent repo is merged already :slightly_smiling_face: IMO you can merge it

File: tdesign/src/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -61,8 +65,8 @@ const CloseableDialogTitle = (props: ICloseableDialogTitleProps) => {
   return (
     <DialogTitle
       sx={{
-        m: 0,
-        p: 2,
+        margin: 0,
```
Comment (onpaws) [inline tdesign/src/Dialog/Dialog.tsx:68]: Nice job RE consistency. I think this was from before we made that decision

File: tdesign/src/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -34,14 +36,16 @@ const Dialog = ({
...
     >
-      <CloseableDialogTitle
-        id={title}
-        onClose={() => {
-          setShowDialog(false);
-        }}
-      >
-        {title}
-      </CloseableDialogTitle>
+      {showCloseIcon && (
```
Comment (onpaws) [inline tdesign/src/Dialog/Dialog.tsx:39]: Nit: if I understand correctly, it seems like this prop shows/hides the entire title bar of the Dialog, not just the CloseIcon. Would we want to call it `showDialogTitle` maybe?

**Comment by onpaws** [comment]
LGTM!

**Comment by onpaws** [comment]
Pretty sure we need to merge this so that the GL job passes. Unless objections going to do that in the next 30 mins or so

**Pull request merged**